diff --git a/esp_flasher_worker.c b/esp_flasher_worker.c
index d47fe05..6145a48 100644
--- a/esp_flasher_worker.c
+++ b/esp_flasher_worker.c
@@ -281,19 +281,19 @@ static int32_t esp_flasher_flash_bin(void* context) {
 }
 
 static void _initDTR(void) {
-    furi_hal_gpio_init(&gpio_ext_pc3, GpioModeOutputPushPull, GpioPullDown, GpioSpeedVeryHigh);
+    furi_hal_gpio_init(&gpio_ext_pc1, GpioModeOutputPushPull, GpioPullDown, GpioSpeedVeryHigh);
 }
 
 static void _initRTS(void) {
-    furi_hal_gpio_init(&gpio_ext_pb2, GpioModeOutputPushPull, GpioPullDown, GpioSpeedVeryHigh);
+    furi_hal_gpio_init(&gpio_ext_pc0, GpioModeOutputPushPull, GpioPullDown, GpioSpeedVeryHigh);
 }
 
 static void _setDTR(bool state) {
-    furi_hal_gpio_write(&gpio_ext_pc3, state);
+    furi_hal_gpio_write(&gpio_ext_pc1, state);
 }
 
 static void _setRTS(bool state) {
-    furi_hal_gpio_write(&gpio_ext_pb2, state);
+    furi_hal_gpio_write(&gpio_ext_pc0, state);
 }
 
 static int32_t esp_flasher_reset(void* context) {
